assets/dev/js/editor/components/template-library/component.js

```diff
@@ -214,21 +214,21 @@ export default class Component extends ComponentModalBase {
 				dialog.show();
 			},
 
-			initDialog: function() {
+			initDialog: function( model ) {
 				InsertTemplateHandler.dialog = elementorCommon.dialogsManager.createWidget( 'confirm', {
 					id: 'elementor-insert-template-settings-dialog',
-					headerMessage: __( 'Import Document Settings', 'elementor' ),
-					message: __( 'Do you want to also import the document settings of the template?', 'elementor' ) + '<br>' + __( 'Attention: Importing may override previous settings.', 'elementor' ),
+					headerMessage: __( 'Apply the settings of this %s too?', 'elementor' ).replace( '%s', model.attributes.type ),
+					message: __( 'This will override the design, layout, and other settings of the %s you’re working on.', 'elementor' ).replace( '%s', elementor.documents.getCurrent().config.type ),
```

Comment:
```suggestion
					message: __( 'This will override the design, layout, and other settings of the %s you’re working on.', 'elementor' ).replace( '%s', elementor.documents.getCurrent().config.post_type_title ),
```

Reply:
Done